### PR TITLE
Integration Tests for Get config

### PR DESCRIPTION
**Pull request by adfost**
Test for configs/{name} endpoint

### CODE REVIEW

File: integration/config_test.go [hunk 1]
```diff
@@ -56,13 +54,65 @@ func Test_ConfigAPI(t *testing.T) {
...
+	tests := []struct {
+		name           string
+		args           args
+		expectedStatus int
+		expectedError  error
+	}{
+		{
+			name: "Get Config - Valid",
+			args: args{
+				api: "/configs/storage_class",
```
Comment (dvaldivia) [inline integration/config_test.go:92]: All test use the same API, just pass the parameter

File: restapi/admin_config.go [hunk 2]
```diff
@@ -43,7 +43,7 @@ func registerConfigHandlers(api *operations.ConsoleAPI) {
 	api.ConfigurationConfigInfoHandler = cfgApi.ConfigInfoHandlerFunc(func(params cfgApi.ConfigInfoParams, session *models.Principal) middleware.Responder {
 		config, err := getConfigResponse(session, params)
 		if err != nil {
-			return cfgApi.NewConfigInfoDefault(int(err.Code)).WithPayload(err)
+			return cfgApi.NewConfigInfoDefault(404).WithPayload(err)
```
Comment (dvaldivia) [inline restapi/admin_config.go:46]: This is invalid, some errors might be 400, please process them according to potential error codes coming from MinIO

File: integration/config_test.go [hunk 3]
```diff
@@ -56,13 +54,65 @@ func Test_ConfigAPI(t *testing.T) {
...
+			}
+			if response != nil {
+				assert.Equal(tt.expectedStatus, response.StatusCode, tt.name+" Failed")
+			}
+		})
+	}
+
+}
+
+func Test_GetConfigAPI(t *testing.T) {
```
Comment (dvaldivia) [inline integration/config_test.go:77]: This is duplicate code from the above test

File: integration/config_test.go [hunk 4]
```diff
@@ -56,13 +54,65 @@ func Test_ConfigAPI(t *testing.T) {
 			client := &http.Client{
```
Comment (dvaldivia) [inline integration/config_test.go:56]: You are instantiating a new client on each loop, this is inefficient, reuse the same client

File: integration/config_test.go [hunk 5]
```diff
@@ -56,13 +54,65 @@ func Test_ConfigAPI(t *testing.T) {
...
 
-			requestDataPolicy := map[string]interface{}{}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			client := &http.Client{
+				Timeout: 3 * time.Second,
+			}
 
-			requestDataJSON, _ := json.Marshal(requestDataPolicy)
```
Comment (dvaldivia) [inline integration/config_test.go:62]: Why ignore the response?

**Review by dvaldivia — CHANGES_REQUESTED** [review]

File: integration/config_test.go (hunk 3 above)
Comment (adfost) [reply integration/config_test.go:77]: No this isn't, the above test lists configs with /configs, this test is /configs/{name}

File: integration/config_test.go (hunk 5 above)
Comment (adfost) [reply integration/config_test.go:62]: The code you are highlighting is unnecessary code that I removed, this was from an endpoint that had a body, and these endpoints don't have a body.

File: integration/config_test.go (hunk 3 above)
Comment (dvaldivia) [reply integration/config_test.go:71]: ok

File: integration/config_test.go [hunk 6]
```diff
@@ -31,6 +29,48 @@ import (
...
+		})
+	}
+
+}
+
+func Test_GetConfigAPI(t *testing.T) {
+	assert := assert.New(t)
+
 	type args struct {
 		api string
```
Comment (dvaldivia) [inline integration/config_test.go:75]: this is no longer the `api` it's the `name` parameter, it should be called that and not include the slash

File: integration/config_test.go (hunk 5 above)
Comment (dvaldivia) [reply integration/config_test.go:62]: this endpoint does have a body, it returns a `configuration`
```yaml
  /configs/{name}:
    get:
      summary: Configuration info
      operationId: ConfigInfo
      parameters:
        - name: name
          in: path
          required: true
          type: string
      responses:
        200:
          description: A successful response.
          schema:
            $ref: "#/definitions/configuration"
        default:
          description: Generic error response.
          schema:
            $ref: "#/definitions/error"
      tags:
        - Configuration
```

**Review by dvaldivia — CHANGES_REQUESTED** [review]

File: integration/config_test.go (hunk 5 above)
Comment (adfost) [reply integration/config_test.go:62]: It doesn't have a body in the request, only in the response. This code that I removed was to set up a json struct for request body.

File: integration/config_test.go [hunk 7]
```diff
@@ -41,13 +79,21 @@ func Test_ConfigAPI(t *testing.T) {
 		expectedError  error
 	}{
 		{
-			name: "Config - Valid",
+			name: "Get Config - Valid",
 			args: args{
-				api: "/configs",
+				name: "/storage_class",
```
Comment (dvaldivia) [inline integration/config_test.go:84]: the `/` is not needed for the config name, that is part of the URL

**Review by dvaldivia — APPROVED** [review]

File: integration/config_test.go [hunk 8]
```diff
@@ -56,12 +102,8 @@ func Test_ConfigAPI(t *testing.T) {
 				Timeout: 3 * time.Second,
 			}
 
-			requestDataPolicy := map[string]interface{}{}
-
-			requestDataJSON, _ := json.Marshal(requestDataPolicy)
-			requestDataBody := bytes.NewReader(requestDataJSON)
 			request, err := http.NewRequest(
-				"GET", fmt.Sprintf("http://localhost:9090/api/v1%s", tt.args.api), requestDataBody)
+				"GET", fmt.Sprintf("http://localhost:9090/api/v1/configs%s", tt.args.name), nil)
```
Comment (bexsoft) [inline integration/config_test.go:106]: Please change this to be 
```suggestion
				"GET", fmt.Sprintf("http://localhost:9090/api/v1/configs/%s", tt.args.name), nil)
```
and remove the slashes from `args.name`

**Review by bexsoft — CHANGES_REQUESTED** [review]

**Review by bexsoft — APPROVED** [review]

**Review by dvaldivia — APPROVED** [review]

**Pull request merged**